File: JavaScript/331-Verify Preorder Serialization of a Binary Tree.js
/**
 * @param {string} preorder
 * @return {boolean}
 */
var isValidSerialization = function(preorder) {
    let slot = 1;
    for (let node of preorder.split(",")){
        slot -= 1;
        if (slot < 0) return false;
        if (node !== '#'){
            slot += 2;
        }
    }
    return slot === 0;
};

File: 331-Verify Preorder Serialization of a Binary Tree.py
class Solution(object):
    def isValidSerialization(self, preorder):
        """
        :type preorder: str
        :rtype: bool
        """
        slot = 1
        
        for node in preorder.split(','):
            slot -= 1
            if slot < 0:
                return False
            if node != '#':
                slot += 2
        
        return slot == 0